frontend/src/app/applications-upload/applications-upload.page.ts

```diff
@@ -0,0 +1,193 @@
+/**
+ * @description
+ * This page send the applications to the server
+ */
+import { HttpClient, HttpHeaders } from '@angular/common/http';
+import { Component, OnInit } from '@angular/core';
+import { NavController } from '@ionic/angular';
+
+import { ToastController } from '../common/ion-wrappers/toast-controller';
+import {
+  EducationInfoEntry,
+  FormsData,
+  KeyCompetenciesEntry,
+} from '../model/forms-data.model';
+import { StorageHandlerService } from '../services/storage-handler.service';
+
+@Component({
+  selector: 'app-applications-upload',
+  templateUrl: './applications-upload.page.html',
+  styleUrls: ['./applications-upload.page.scss'],
+})
+export class ApplicationsUploadPage implements OnInit {
+  constructor(
+    private navController: NavController,
+    private storage: StorageHandlerService,
+    private httpClient: HttpClient,
+    private toastController: ToastController
+  ) {}
+  /**
+   * The total application size.
+   */
+  totalSize = 0;
+  /**
+   * The application size that is being uploaded.
+   */
+  uploadSize = 1;
+  /**
+   * To check whether all the applications are uploaded.
+   */
+  isSuccess = false;
+  /**
+   * To check whether there is error during upload.
+   */
+  isError = false;
+  /**
+   * To display the error message dynamically.
+   */
+  errorMessage: string;
+  /**
+   * URL of the host.
+   */
+  url = 'http://localhost:9090';
+
+  /**
+   * In this method locally stored data are fetched and post request function is called for one application at time.
+   */
+  ngOnInit(): void {
+    this.storage
+      .getAllItems<FormsData>(this.storage.applicantDetailsDb)
+      .then(async (data) => {
+        const applicantDetailsList = data.filter((x) => x.isRated === 1);
+        this.totalSize = applicantDetailsList.length;
+        for (const x of applicantDetailsList) {
+          if (!this.isError) {
```

Comment:
Is this check needed? Let's say there are 100 records. And 1st API fails for some reason, then isError will become true. And rest of the API which were yet to be sent to the method won't be sent.

Reply:
ya thats true but now the error is more genera. If one application fail, all the corresponding applications fail. We can change this with more specific errors in the coming sprint.